### PR TITLE
add v2 wallet settings page

### LINKED ISSUE #1033: Implement Decred specific settings
Make a settings page for decred specific wallets based on the figma:

https://www.figma.com/file/e0cYaAzkXyXXPY00v23cQY/GoDCR-%26-DCRMobile?node-id=4659%3A182253

### PR DESCRIPTION
**Pull request by Sirmorrison**
This Pr Fix #1033 
This Pr implement the V2 dcr wallet settings design and all its functionalities.

image

### CODE REVIEW

**Comment by devchoplife** [comment]
Regarding the description, I think closes is a better verb because the issue is not a bug.

File: ui/page/security/validate_address.go [hunk 1]
```diff
@@ -32,8 +33,7 @@ type ValidateAddressPage struct {
 	addressEditor         decredmaterial.Editor
 	clearBtn, validateBtn decredmaterial.Button
 	stateValidate         int
-	walletName            string
-	isEnabled             bool
+	_                     string
```
Comment (devchoplife) [inline ui/page/security/validate_address.go]: This variable was declared but not used. Also, I am not sure why it was created.

File: ui/values/localizable/es.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ const ES = `
 "receive" = "Recibir";
 "online" = "En línea, ";
 "offline" = "Desconectado, ";
-"showDetails" = "Mostrar detalles";
+"viewDetails" = "View details";
```
Comment (devchoplife) [inline ui/values/localizable/es.go:15]: This should be translated to spanish

File: ui/values/localizable/es.go [hunk 3]
```diff
@@ -90,7 +90,8 @@ const ES = `
 "viewProperty" = "Ver propiedad";
 "stakeShuffle" = "StakeShuffle";
 "rename" = "Renombrar";
-"renameWalletSheetTitle" = "Renombrar la cartera";
+"renameWalletSheetTitle" = "Change wallet name";
+"securityToolsInfo" = "%v Various tools that help in different aspects of crypto currency security will be located here. %v"
```
Comment (devchoplife) [inline ui/values/localizable/es.go:95]: This should be translated to spanish

File: ui/values/localizable/es.go [hunk 4]
```diff
@@ -478,4 +479,8 @@ syncCompTime" = "Est. Sync completion time"
 "selectDexServerToOpen" = "Select the Dex server you would like to open."
 "addDexServer" = "Add dex server"
 "canBuy" = "Can Buy"
+"removeWalletInfo" = "%v Are you sure you want to remove %v %s%v? Enter the name of the wallet below to verify. %v"
+"propNotif" = "Proposal notification"
+"peer" = "Peer"
+"confirmUmixedSpending" = "Confirm to allow spending from unmixed accounts"
```
Comment (devchoplife) [inline ui/values/localizable/es.go:487]: This should be translated to spanish

File: ui/values/localizable/fr.go [hunk 5]
```diff
@@ -24,7 +24,7 @@ const FR = `
 "resumeAccountDiscoveryTitle" = "Déverouiller pour reprendre la restauration";
 "unlockWithPassword" = "Déverrouiller avec mot de passe"
 "offline" = "Hors ligne, ";
-"showDetails" = "Montrer les détails";
+"viewDetails" = "View details";
```
Comment (devchoplife) [inline ui/values/localizable/fr.go:27]: This should be translated to French

File: ui/values/localizable/fr.go [hunk 6]
```diff
@@ -50,7 +50,8 @@ const FR = `
 "blockHeaderFetchedCount" = "%d de %d";
 "timeLeft" = "%v restant";
 "currencyConversion" = "Conversion de devise";
-"renameWalletSheetTitle" = "Renommer le porte-monnaie";
+"renameWalletSheetTitle" = "Change wallet name";
+"securityToolsInfo" = "%v Various tools that help in different aspects of crypto currency security will be located here. %v"
```
Comment (devchoplife) [inline ui/values/localizable/fr.go:54]: This should be translated to French

File: ui/values/localizable/fr.go [hunk 7]
```diff
@@ -455,4 +456,8 @@ const FR = `
 "selectDexServerToOpen" = "Select the Dex server you would like to open."
 "addDexServer" = "Add dex server"
 "canBuy" = "Can Buy"
+"removeWalletInfo" = "%v Are you sure you want to remove %v %s%v? Enter the name of the wallet below to verify. %v"
+"propNotif" = "Proposal notification"
+"peer" = "Peer"
+"confirmUmixedSpending" = "Confirm to allow spending from unmixed accounts"
```
Comment (devchoplife) [inline ui/values/localizable/fr.go:464]: This should be translated to French

**Review by devchoplife — CHANGES_REQUESTED** [review]
- The page content width should be reduced as seen on the mockup:
image

vs 
image

**Review by dreacot — DISMISSED** [review]
1. When disabling fetching of proposals, proposals notifications should also default to being turned off
Screenshot from 2022-08-16 19-47-58

2. clicking on "Check wallet log" crashes the app. see log below;

```
2022/08/16 19:49:05 FATAL -- failed to create Watcher
goroutine 322 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/nxadm/tail/util.Fatal({0x17e8c8b, 0xc8b90b}, {0x0, 0x0, 0x0})
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/util/util.go:22 +0x97
github.com/nxadm/tail/watch.(*InotifyTracker).run(0xc006e10600)
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:219 +0x68
created by github.com/nxadm/tail/watch.glob..func1
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:54 +0x173
exit status 1
```
3. The modal below should say "OK" based on the mockup, it also looks like there is excessive top and bottom spacing. on second thought, due to the vast amount of modals that requiring changing, this would be better suited in it's own PR
Screenshot from 2022-08-16 19-52-39
Screenshot from 2022-08-16 19-53-47

4.

File: ui/values/localizable/es.go (hunk 2 above)
Comment (dreacot) [reply ui/values/localizable/es.go:15]: it doesn't necessarily have to, we would have a translator do all non-english translations

File: ui/values/localizable/es.go (hunk 4 above)
Comment (dreacot) [reply ui/values/localizable/es.go:487]: it doesn't necessarily have to, we would have a translator do all non-english translations

File: ui/values/localizable/fr.go (hunk 6 above)
Comment (dreacot) [reply ui/values/localizable/fr.go:54]: it doesn't necessarily have to, we would have a translator do all non-english translations

File: ui/values/localizable/fr.go (hunk 5 above)
Comment (dreacot) [reply ui/values/localizable/fr.go:27]: it doesn't necessarily have to, we would have a translator do all non-english translations

File: ui/values/localizable/fr.go (hunk 7 above)
Comment (dreacot) [reply ui/values/localizable/fr.go:464]: it doesn't necessarily have to, we would have a translator do all non-english translations

**Comment by Sirmorrison** [comment]
> Regarding the description, I think closes is a better verb because the issue is not a bug.

i don't really think it matters... the idea is to link the pr to the issue and it does just that..

**Comment by Sirmorrison** [comment]
> * The page content width should be reduced as seen on the mockup:
>       image
> 
> 
> vs image

the mockup shows the design in full screen... kindly open the app in full display and compare. I would however add the indentation on the left.

File: ui/values/localizable/es.go (hunk 2 above)
Comment (Sirmorrison) [reply ui/values/localizable/es.go:15]: The idea is to have a place holder for the other languages pending when we do a general sting localization like the #906

File: ui/values/localizable/es.go (hunk 3 above)
Comment (Sirmorrison) [reply ui/values/localizable/es.go:95]: The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

File: ui/values/localizable/es.go (hunk 4 above)
Comment (Sirmorrison) [reply ui/values/localizable/es.go:487]: The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

File: ui/values/localizable/fr.go (hunk 5 above)
Comment (Sirmorrison) [reply ui/values/localizable/fr.go:27]: The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

File: ui/values/localizable/fr.go (hunk 6 above)
Comment (Sirmorrison) [reply ui/values/localizable/fr.go:54]: The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

File: ui/values/localizable/fr.go (hunk 7 above)
Comment (Sirmorrison) [reply ui/values/localizable/fr.go:464]: The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

**Comment by Sirmorrison** [comment]
@dreacot i could not reproduce this. what where your repro steps? also the log does not point to any godcr file.

2022/08/16 19:49:05 FATAL -- failed to create Watcher
goroutine 322 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/nxadm/tail/util.Fatal({0x17e8c8b, 0xc8b90b}, {0x0, 0x0, 0x0})
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/util/util.go:22 +0x97
github.com/nxadm/tail/watch.(*InotifyTracker).run(0xc006e10600)
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:219 +0x68
created by github.com/nxadm/tail/watch.glob..func1
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:54 +0x173
exit status 1

**Comment by devchoplife** [comment]
> @dreacot i could not reproduce this. what where your repro steps? also the log does not point to any godcr file.
> 
> 2022/08/16 19:49:05 FATAL -- failed to create Watcher goroutine 322 [running]: runtime/debug.Stack() /usr/local/go/src/runtime/debug/stack.go:24 +0x65 github.com/nxadm/tail/util.Fatal({0x17e8c8b, 0xc8b90b}, {0x0, 0x0, 0x0}) /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/util/util.go:22 +0x97 github.com/nxadm/tail/watch.(*InotifyTracker).run(0xc006e10600) /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:219 +0x68 created by github.com/nxadm/tail/watch.glob..func1 /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:54 +0x173 exit status 1

I cannot reproduce the error too

**Comment by JustinBeBoy** [comment]
1. add notice when sign message success like mockup and remove background of "signature" text
image

2. "DCR" not "DC"
aaaaaa

3. Seem like modal to create new account of wallet ui not like mockup
app:
bb1
vs
mockup:
bb2

4. modal info of verify message and sign message not like mockup
cccc
vs 
cccc2

File: ui/modal/text_input_modal.go [hunk 8]
```diff
@@ -86,17 +86,17 @@ func (tm *TextInputModal) SetTextWithTemplate(template string) *TextInputModal {
...
 		tm.textCustomTemplate = allowUnspendUnmixedAcct(tm.Load)
+	case RemoveWalletInfoTemplate:
+		tm.textCustomTemplate = removeWalletInfo(tm.Load)
 	}
 	return tm
 }
 
 func (tm *TextInputModal) Handle() {
 
 	if editorsNotEmpty(tm.textInput.Editor) {
```
Comment (JustinBeBoy) [inline ui/modal/text_input_modal.go]: it should be `tm. tm.isEnabled = editorsNotEmpty(tm.textInput.Editor)`

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]

**Comment by Sirmorrison** [comment]
* add notice when sign message success like mockup and remove background of "signature" text
image
There is a Pr https://github.com/planetdecred/godcr/pull/1039 already to resolve this no need implementing this twice.

File: ui/page/wallet_settings_page.go [hunk 9]
```diff
@@ -65,39 +100,67 @@ func NewWalletSettingsPage(l *load.Load, wal *dcrlibwallet.Wallet) *WalletSettin
...
-					pg.stakeshuffle(),
+					pg.securityTools(),
 					pg.debug(),
 					pg.dangerZone(),
 				}
 
 				return pg.pageContainer.Layout(gtx, len(w), func(gtx C, i int) D {
-					return w[i](gtx)
+					return layout.Inset{Left: values.MarginPadding50}.Layout(gtx, w[i])
+					// return w[i](gtx)
```
Comment (JustinBeBoy) [inline ui/page/wallet_settings_page.go]: remove it

File: ui/page/wallet_settings_page.go [hunk 10]
```diff
@@ -370,83 +574,180 @@ func (pg *WalletSettingsPage) HandleUserInteractions() {
...
+func (pg *WalletSettingsPage) resetDexDataModal() {
+	// Show confirm modal before resetting dex client data.
+	confirmModal := modal.NewInfoModal(pg.Load).
+		Title(values.String(values.StrConfirmDexReset)).
+		Body(values.String(values.StrDexResetInfo)).
+		NegativeButton(values.String(values.StrCancel), func() {}).
+		NegativeButtonStyle(pg.Theme.Color.Primary, pg.Theme.Color.Surface).
+		PositiveButton(values.String(values.StrResetDexClient), func(isChecked bool) bool {
+			if pg.Dexc().Reset() {
+				pg.Toast.Notify("DEX client data reset complete.")
```
Comment (JustinBeBoy) [inline ui/page/wallet_settings_page.go:736]: localizable

File: ui/page/wallet_settings_page.go [hunk 11]
```diff
@@ -370,83 +574,180 @@ func (pg *WalletSettingsPage) HandleUserInteractions() {
...
+			} else {
+				pg.Toast.NotifyError("DEX client data reset failed. Check the logs.")
```
Comment (JustinBeBoy) [inline ui/page/wallet_settings_page.go:738]: localizable

**Comment by JustinBeBoy** [comment]
1 nothing show when change account name successful


2. please remove back button

**Review by JustinBeBoy — APPROVED** [review]

**Pull request merged**